From PeterHuewe@gmx.de  Thu Jan 14 16:58:07 2010
From: Peter Huewe <PeterHuewe@gmx.de>
Date: Thu, 7 Jan 2010 20:03:51 +0100
Subject: Staging: asus_oled: Add NULL test for kmalloc
To: "Greg Kroah-Hartman" <gregkh@suse.de>
Cc: Andre Haupt <andre@bitwigglers.org>, "Kevin A. Granade" <kevin.granade@gmail.com>, linux-kernel@vger.kernel.org, devel@driverdev.osuosl.org, sjakub@gmail.com
Message-ID: <201001072003.51722.PeterHuewe@gmx.de>


From: Peter Huewe <peterhuewe@gmx.de>
Date: Thu, 7 Jan 2010 19:57:36 +0100

This patch adds a NULL test to check wether kmalloc was successful or
not.

Signed-off-by: Peter Huewe <peterhuewe@gmx.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/asus_oled/asus_oled.c |    5 +++++
 1 file changed, 5 insertions(+)

--- a/drivers/staging/asus_oled/asus_oled.c
+++ b/drivers/staging/asus_oled/asus_oled.c
@@ -430,6 +430,11 @@ static ssize_t odev_set_picture(struct a
 
 		kfree(odev->buf);
 		odev->buf = kmalloc(odev->buf_size, GFP_KERNEL);
+		if (odev->buf == NULL) {
+			odev->buf_size = 0;
+			printk(ASUS_OLED_ERROR "Out of memory!\n");
+			return -ENOMEM;
+		}
 
 		memset(odev->buf, 0xff, odev->buf_size);
 
